core/src/main/java/org/apache/iceberg/BaseScan.java

```diff
@@ -59,6 +60,23 @@
   private static final List<String> SCAN_WITH_STATS_COLUMNS =
       ImmutableList.<String>builder().addAll(SCAN_COLUMNS).addAll(STATS_COLUMNS).build();
 
+  protected static final List<String> DELETE_SCAN_COLUMNS =
+      ImmutableList.of(
+          "snapshot_id",
+          "content",
```

Comment:
This has to differ from data file's SCAN_COLUMNS by including content, which is used in the scan to filter later.